pkg/sosreport/index.js

```diff
@@ -175,6 +176,14 @@ function init() {
         cockpit.translate();
         $('body').prop("hidden", false);
 
+        function update_admin_allowed() {
+            $("#switch-instructions").toggle(superuser.allowed === false);
+            $("#create-button").toggle(!!superuser.allowed);
```

Comment:
Any reason for the two different styles? Ideally the value should be exactly true or false, not null here. But if it ever *is* `null`, then the first line is wrong -- we don't want to show the superuser warning if the page is still initializing and the sudo status isn't known yet. So for consistency I'd just  use the second form always.

Reply:
`update_admin_allowed` can be called at any time, not just from a "changed" handler, so `superuser.allowed` will indeed be `null` here routinely.
I think `superuser.allowed === false` is correct. It evaluates to `false` when `superuser.allowed` is null and `toggle(false)` means to hide the element.
`!!superuser.allowed` could be written as `superuser.allowed === true` but I was thinking that maybe we get more truthy values for the "allowed" field in the future and then we would erroneously hide the button.